fieldservice_stage_server_action/models/fsm_equipment.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class FSMEquipment(models.Model):
+    _inherit = "fsm.equipment"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        records = super().create(vals_list)
+        for record, vals in zip(records, vals_list):
+            action = record.stage_id.action_id
+            if vals.get("stage_id") and action:
```

Comment:
```suggestion
        for record in records:
            action = record.stage_id.action_id
            if action:
```

Checking for `vals['stage_id']` seems unnecessary here.
As you're creating records, you don't need to check vals for updates.. simply see if you have an action on the stage_id.